src/build/buildNativefierApp.ts

```diff
@@ -1,7 +1,9 @@
 import * as path from 'path';
 
+import * as asar from 'asar';
```

Comment:
Same as for `fs-extra`: if using a new third-party dep, you need to declare the extra dep in package.json.

But wait, all this is strange. Here you're doing `asar` packaging on your own, in Nativefier. We're not supposed to, Nativefier shouldn't have to re-implement it, it's already handled by electron-packager! https://electron.github.io/electron-packager/master/interfaces/electronpackager.options.html#asar . That's why right now in current Nativefier there basically zero code _do_ the packaging; normally we simply pass the `asar` option to electron-packager and it does the packaging for us.

Now, obviously, it's broken now, but rather than doing it here in Nativefier, I'd like to instead

1. Understand why it's broken now, maybe we're mis-using electron-packager or doing something it doesn't like
2. Reach out to electron-packager for help if necessary
3. If something's broken in electron-packager, offer a patch to them.

Please look at these.

Reply:
The issue stems from the fact that `copyIconsIfNecessary()' occurs after `electronPackager()` completes.

As far as I can tell, there is no way to perform the copy action without the path generated from `electronPackager()`